File: Interface.py
from abc import ABC, abstractclassmethod
import abc

class Player(ABC):
    def __init__(self):
        pass

    #metodo abtracto 
    @abstractclassmethod
    def open(self, filepath):
        pass

    @abstractclassmethod
    def play(self):
        pass

    @abstractclassmethod
    def stop(self):
        pass

    @abstractclassmethod
    def setvolume(self, value):
        pass

#*******************************************************************                                  

class PlayerPitchable(ABC):
    def __init__(self):
        pass
    
    @abstractclassmethod
    def setPitch(self, pitch):
        pass

#*******************************************************************
class VLC(Player, PlayerPitchable):

    def __init__(self):
        self.isOpen = True
        self.isPlaying = False
        self.volume = 0
        print("The VLC constructor was invoked.")

    # Metodo abtracto que debo implementar en todas las clases derivadas 
    def open(self, filepath):
        self.filePath = filepath
        if self.isOpen:
            print("The audiofile: " + self.filePath + " is open.")

    def setPitch(self, value):   
        self.pitch = value
        print("The pitch value is: " , self.pitch)

    def setvolume(self, value):
        self.volume = value
        print("The volume value is: " , self.volume)

    def play(self):
        if self.isOpen:
            self.isPlaying = True
            print("The audiofile is playing.")

    def stop(self):
        if self.isPlaying:
            self.isPlaying = False
            print("The audiofile is stopped.")

#********************************************************************

vlcPlayer = VLC()
vlcPlayer.open("./resources/orchestral.ogg")
vlcPlayer.play()
vlcPlayer.setvolume(13)

File: Interface.java
public interface Player
{
    
    public void open(String filePath);
    public void play();
    public void stop();
    public void setVolume(float value);
};
//PlayerPitchable Interface:
public interface PlayerPitchable
{
public void setPitch(float value);
};

//VLC concrete class
public class VLC  implements Player , PlayerPitchable
{
    protected Boolean isOpen = false;
    protected Boolean isPlaying = false;
    protected float volume;
    protected float pitch;

    public VLC()
    {
        System.out.println("The VLC constructor was invoked.");
    }

    @Override
    public void open(String filePath)
    {
        //@ToDo: por ahora simulamos la apertura correcta
        this.isOpen = true;
        System.out.println("The audiofile: " + filePath + " is open.");
    }
    @Override
    public void play()
    {
        //@ToDo: por ahora simulamos la reproducción correcta
        if(this.isOpen) 
            this.isPlaying = true;
            System.out.println("The audiofile is playing.");
    }
    @Override
    public void stop()
    {
        //@ToDo: por ahora simulamos la detención correcta
        if(this.isPlaying) 
            this.isPlaying = false;
            System.out.println("The audiofile is stopped.");
    }
    @Override
    public void setVolume(float value)
    {
        this.volume = value;
        System.out.println("The volume value is: " + this.volume);
    }
    @Override
    public void setPitch(float value)
    {
        this.pitch = value;
        System.out.println("The pitch value is: " + this.pitch);
    }

}

public static void main(String[] args)
{
    System.out.println("--------");
    VLC vlcPlayer = new VLC();
    vlcPlayer.open("./resources/orchestral.ogg");
    vlcPlayer.play();
    vlcPlayer.setVolume(13);
    System.out.println("--------");
}
